src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt

```diff
@@ -95,14 +96,23 @@ class RecallNotificationContextFactoryTest {
 
   private fun prevConsMainNameOptions(): Stream<Arguments> {
     return Stream.of(
-      Arguments.of(true, "Bobby Badger", "Bobby Badger"),
-      Arguments.of(false, "", "Bobbie Badger")
+      Arguments.of(true, "Bobby Badger", null, "Bobby Badger"),
+      Arguments.of(true, "Bobby Badger", PreConsMainNameCategory.FIRST_LAST, "Bobbie Badger"),
+      Arguments.of(true, "Bobby Badger", PreConsMainNameCategory.FIRST_MIDDLE_LAST, "Bobbie Bertie Badger"),
+      Arguments.of(true, "Bobby Badger", PreConsMainNameCategory.OTHER, "Bobby Badger"),
+      Arguments.of(false, "", null, "Bobbie Badger"),
+      Arguments.of(false, "", PreConsMainNameCategory.FIRST_LAST, "Bobbie Badger"),
+      Arguments.of(false, "", PreConsMainNameCategory.FIRST_MIDDLE_LAST, "Bobbie Bertie Badger"),
+      Arguments.of(false, "Other Name", PreConsMainNameCategory.OTHER, "Other Name"),
+      Arguments.of(null, "", PreConsMainNameCategory.FIRST_LAST, "Bobbie Badger"),
+      Arguments.of(null, "", PreConsMainNameCategory.FIRST_MIDDLE_LAST, "Bobbie Bertie Badger"),
+      Arguments.of(null, "", PreConsMainNameCategory.OTHER, ""),
     )
   }
 
-  @ParameterizedTest(name = "create RecallSummaryContext when hasPreviousConvictionMainName is {0}")
+  @ParameterizedTest(name = "create RecallSummaryContext when hasPreviousConvictionMainName is {0} & category is {2}")
   @MethodSource("prevConsMainNameOptions")
-  fun `create RecallSummaryContext with required details`(hasPrevConsMainName: Boolean, prevConsMainName: String, policeFileName: String) {
+  fun `create RecallSummaryContext with required details`(hasPrevConsMainName: Boolean?, prevConsMainName: String?, prevConsMainNameCategory: PreConsMainNameCategory?, expectedName: String) {
```

Comment:
Again I know it was like this before but repeating this whole test for each of these scenarios to me obscures both purposes: can we split this so only a much smaller focused test is repeated for the scenarios?  Ta.

Reply:
It's being cleared up in the next PR once we remove the boolean